homeassistant/components/google_assistant/trait.py

```diff
@@ -1654,27 +1670,68 @@ async def _execute_set_volume(self, data, params):
         )
 
     async def _execute_volume_relative(self, data, params):
-        # This could also support up/down commands using relativeSteps
-        relative = params["volumeRelativeLevel"]
-        current = self.state.attributes.get(media_player.ATTR_MEDIA_VOLUME_LEVEL)
+        relative = params["relativeSteps"]
```

Comment:
You need both relativeSteps and volumeRelativeLevel here. There is no guarantee that steps are related to level change by steps / 100.

Grab the relative value inside the different supported cases.

Reply:
I think that was how a previous version of the API worked, but it's not how it's documented to work now. The documentation for the [volume trait](https://developers.google.com/assistant/smarthome/traits/volume) now has everything being relative to the `volumeMaxLevel` you set in the `SYNC` response. That applies to both the `setVolume` and `volumeRelative` commands:

> ## action.devices.commands.setVolume
> Set volume to the requested level, based on `volumeMaxLevel`.

> ## action.devices.commands.volumeRelative
> Set volume up or down _n_ steps, based on `volumeMaxLevel`.

Since we specify that `volumeMaxLevel` is 100 in our SYNC response, we can safely assume that both absolute and relative volume commands are out of 100.

FWIW, the `volumeRelativeLevel` field isn't even documented as being sent any more, just `relativeSteps`.

Reply:
Huh. You seem to be right. The  I'm okey with this.